scripts/populate_institutions.py

```diff
@@ -44,11 +54,40 @@ def update_or_create(inst_data):
         return inst, True
 
 
-def main(env):
-    INSTITUTIONS = []
+def main():
+
+    args = parser.parse_args()
+    server_env = args.env
+    update_ids = args.ids
+    update_all = args.all
+
+    if not server_env or server_env not in ENVS:
+        logger.error('A valid environment must be specified: {}'.format(ENVS))
+        sys.exit(1)
+    institutions = INSTITUTIONS[server_env]
+
+    if not update_all and not update_ids:
+        logger.error('Nothing to update or create. Please either specify a list of institutions '
+                     'using --id or run for all with --all')
+        sys.exit(1)
+    elif update_all:
+        institutions_to_update = institutions
+    else:
+        institutions_to_update = [inst for inst in institutions if inst['_id'] in update_ids]
+        if len(institutions_to_update) < len(update_ids):
+            logger.warning('One or more institutions provided via -i or --ids do not match any '
```

Comment:
Find the non-existing ID(s) is a waste of time (if not too expensive) and **break my beautiful list comprehension** since the `institutions` is a list instead of `dict` (I am not going to change it). Just comparing the length and informing the users that they might have mistyped the ID(s) is good enough. The log shows what institutions have been updated anyway.

Reply:
You can keep your comprehension without sacrificing error handling
```
In [1]: abc = {'a': 'a', 'b': 'b', 'c': 'c'}

In [2]: try:
   ...:     [abc[a] for a in ['a', 'b', 'd']]
   ...: except KeyError:
   ...:     print('ASDF')
   ...:     raise
   ...:
ASDF
```

Reply:
Good point, will do this.

Reply:
Here is the improvement that I made without using `try ... except ...`. Now the script will error if invalid institution IDs are provided.

```python
institutions_to_update = [inst for inst in institutions if inst['_id'] in update_ids]
diff_list = list(set(update_ids) - set([inst['_id'] for inst in institutions_to_update]))
if diff_list:
    logger.error('One or more institution ID(s) provided via -i or --ids do not match any '
                 'existing records: {}.'.format(diff_list))
    sys.exit(1)      
```